src/automation.js

```diff
@@ -0,0 +1,97 @@
+import { rdfs, rdf } from '@tpluscode/rdf-ns-builders'
+import $rdf from '@rdfjs/dataset'
+import clownface from 'clownface'
+import TermMap from '@rdfjs/term-map'
+import * as dimensionQueries from './queries/dimensions.js'
+import * as ns from './ns.js'
+
+export async function generateDimensions(view, queries = dimensionQueries) {
+  const sources = view.out(ns.ssz.source)
+
+  clearGeneratedDimensions(view)
+
+  await createMeasureDimensions(view, sources, queries)
+  await createKeyDimensions(view, sources, queries)
+
+  return clownface({
+    dataset: $rdf.dataset([...view.dataset]),
+    term: view.term,
+  })
+}
+
+async function createKeyDimensions(view, sources, { findKeyDimensions }) {
+  const results = await findKeyDimensions(sources.out(ns.view.cube).terms)
+
+  const dimensions = results.reduce((map, { cube, dimension, label }) => {
+    const value = map.get(dimension) || { label, sources: [] }
+    value.sources.push(sources.has(ns.view.cube, cube).term)
+
+    map.set(dimension, value)
+    return map
+  }, new TermMap())
+
+  for (const [dimension, { label, sources }] of dimensions) {
+    view.addOut(ns.view.dimension, viewDim => {
+        viewDim
+          .addOut(rdf.type, ns.view.Dimension)
+          .addOut(ns.ssz.generated, true)
```

Comment:
Sets a flag to remove the dimension when regenerating. See `clearGeneratedDimensions` below